ArduCopter/mode_auto.cpp

```diff
@@ -411,7 +411,7 @@ void Copter::ModeAuto::land_run()
     _copter.land_run_vertical_control();
 }
 
-bool Copter::ModeAuto::landing_gear_should_be_deployed()
+bool Copter::ModeAuto::landing_gear_should_be_deployed() const
 {
     switch(_mode) {
     case Auto_Land:
```

Comment:
GitHub doesn't allow me to comment below. Wouldn't it be better in Auto_RTL to do `return _copter.mode_rtl.landing_gear_should_be_deployed()`?

Reply:
Not really related to this PR.

But still a great cleanup.  Added a patch on top.